src/pages/Swap/index.tsx

```diff
@@ -412,6 +459,38 @@ export default function Swap() {
 
   const priceImpactTooHigh = priceImpactSeverity > 3 && !isExpertMode
 
+  // Handle time based logging events and event properties.
+  useEffect(() => {
+    const now = new Date()
+    // If a trade exists, and we need to log the receipt of this new swap quote:
+    if (newSwapQuoteNeedsLogging && !!trade) {
+      // Set the current datetime as the time of receipt of latest swap quote.
+      setSwapQuoteReceivedDate(now)
+      // Log swap quote.
+      sendAnalyticsEvent(
+        EventName.SWAP_QUOTE_RECEIVED,
+        formatAnalyticsEventProperties(trade, fetchingSwapQuoteStartTime)
+      )
+      // Latest swap quote has just been logged, so we don't need to log the current trade anymore
+      // unless user inputs change again and a new trade is in the process of being generated.
+      setNewSwapQuoteNeedsLogging(false)
+      // New quote is not being fetched, so set start time of quote fetch to undefined.
+      setFetchingSwapQuoteStartTime(undefined)
+    }
+    // If another swap quote is being loaded based on changed user inputs:
```

Comment:
i'm not following this comment

Reply:
when routeIsLoading is true, a new swap quote (that we should log) is being generated because user inputs have changed.